addons/website/views/website_templates.xml

```diff
@@ -1459,7 +1459,7 @@
                             <t t-set="_dropdown_menu_class" t-valuef="dropdown-menu-end"/>
                         </t>
                         <!-- Call To Action -->
-                        <t t-call="website.header_call_to_action_pill"/>
+                        <t t-call="website.header_call_to_action"/>
```

Comment:
I think that in this case, you need to t-call `placeholder_header_call_to_action` otherwise it crashes if the `header_call_to_action` option is disabled? I think this was the whole point of having `placeholder_XXX` ones :wink: I actually was surprised your `header_call_to_action_XXX` ones worked when I tested: funny you cannot t-call an inactive template but you can t-call one which depends on an inactive one (and if it is inactive, the t-call then does nothing). I was actually worried this is relying a bit too much on an implementation detail.

If we remove the `pill` one (which was the most called one), could we also remove the other and replace them with `t-set`?

Instead of
```xml
<t t-call="website.header_call_to_action_large"/>
```
Doing:
```xml
<t t-call="website.placeholder_header_call_to_action">
   <t t-set="type" t-valuef="large"/>
</t>
```

That way we don't rely on an implementation detail (possibly even a bug) and we actually advertise the fact those can be disabled (placeholder_XXX).

Reply:
> you need to t-call placeholder_header_call_to_action otherwise it crashes if the header_call_to_action option is disabled? I think this was the whole point of having placeholder_XXX ones

I can do that, although AFAIK I've been toggling on/off the CTA, switching headers in between, and I don't see anything crashing without the placeholder.

> If we remove the pill one (which was the most called one), could we also remove the other and replace them with t-set?

I've been trying to, but this means adding `t-att(f)-` attributes on the CTA button and its container, which turns them into fixed arch views. 1) It removes the options for the button (animations, modifying the link, etc.) and 2) the `customCtaStyle()` fix definitely does not work afterwards.

Reply:
Following IRL discussion too.

> I can do that, although AFAIK I've been toggling on/off the CTA, switching headers in between, and I don't see anything crashing without the placeholder.

I investigated a bit to see if I was crazy or not. It seems indeed that placeholder template are not needed anymore but they were needed at the beginning. Found the first commit which introduced an empty placeholder: https://github.com/odoo/odoo/commit/9b2178952df9b0001e884731d744145f18fbe061-> at that point, it crashes if you make the `header_language_selector` a base view (no xpath etc), and you try to t-call it when it is disabled. I would be curious to find what "fixed" that :+1: 

For 17.1, I'll create a task. For 17.0, let's keep placeholder views for now.

> I've been trying to, but this means adding t-att(f)- [...]

IRL discussion result: we keep the the xpath but we remove all default style which conflict with edition